auslib/migrate/versions/021_add_required_signoffs.py

```diff
@@ -0,0 +1,187 @@
+        "product_req_signoffs_scheduled_changes", metadata,
+        Column("sc_id", Integer, primary_key=True, autoincrement=True),
+        Column("scheduled_by", String(100), nullable=False),
+        Column("complete", Boolean, default=False),
+        Column("change_type", String(50), nullable=False),
+        Column("data_version", Integer, nullable=False),
+        Column("base_product", String(15), nullable=False),
+        Column("base_channel", String(75), nullable=False),
+        Column("base_role", String(50), nullable=False),
+        Column("base_signoffs_required", Integer),
+        Column("base_data_version", Integer),
+    )
+
+    product_required_signoffs_scheduled_changes_history = Table(
+        "product_req_signoffs_scheduled_changes_history", metadata,
+        Column("change_id", Integer, primary_key=True, autoincrement=True),
+        Column("changed_by", String(100), nullable=False),
+        Column("sc_id", Integer, nullable=False, autoincrement=True),
+        Column("scheduled_by", String(100)),
+        Column("complete", Boolean, default=False),
+        Column("change_type", String(50)),
+        Column("data_version", Integer),
+        Column("base_product", String(15)),
+        Column("base_channel", String(75)),
+        Column("base_role", String(50)),
+        Column("base_signoffs_required", Integer),
+        Column("base_data_version", Integer),
+    )
+    product_required_signoffs_scheduled_changes_history.append_column(Column("timestamp", bigintType, nullable=False))
+
+    product_required_signoffs_scheduled_changes_conditions = Table(
+        "product_req_signoffs_scheduled_changes_conditions", metadata,
+        Column("sc_id", Integer, primary_key=True, autoincrement=True),
+        Column("data_version", Integer),
+    )
+    product_required_signoffs_scheduled_changes_conditions.append_column(Column("when", bigintType))
+
+    product_required_signoffs_scheduled_changes_conditions_history = Table(
+        "product_req_signoffs_scheduled_changes_conditions_history", metadata,
+        Column("change_id", Integer, primary_key=True, autoincrement=True),
+        Column("changed_by", String(100), nullable=False),
+        Column("sc_id", Integer, nullable=False),
+        Column("data_version", Integer),
+    )
+    product_required_signoffs_scheduled_changes_conditions_history.append_column(Column("timestamp", bigintType, nullable=False))
+    product_required_signoffs_scheduled_changes_conditions_history.append_column(Column("when", bigintType))
+
+    product_required_signoffs_scheduled_changes_signoffs = Table( # noqa
+        "product_req_signoffs_scheduled_changes_signoffs", metadata,
+        Column("sc_id", Integer, primary_key=True, autoincrement=False),
+        Column("username", String(100), primary_key=True),
+        Column("role", String(50), nullable=False),
+    )
+
+    product_required_signoffs_scheduled_changes_signoffs_history = Table( # noqa
```

Comment:
Yeah, I'm not particularly happy with some of these names =\. We're actually bumping up against the default mysql table name length in some cases (which is why "required" became "req"). I'm not sure I'm a big fan of double underscore as a special delimeter, but I agree that a special delimeter would be helpful. I wonder if mysql supports emoji...

Is it just the database names that throw you? Eg: are references like dbo.productRequiredSignoffs.scheduled_changes.signoffs.history also confusing?